zulip/integrations/slack/slackdata2zulipdata.py

```diff
@@ -139,20 +141,33 @@ def channels2zerver_stream(slack_dir, realm_id, added_users):
             id=stream_id_count)
         zerver_stream.append(stream)
         added_channels[stream['name']] = stream_id_count
+       
+        # construct the recipient object and append it zerver_recipient
+        recipient = dict(
+             type_id=stream_id_count,
+             id=stream_id_count,
+             type=2)
+        zerver_recipient.append(recipient)
+        # TOODO add recipients for private message and huddles
 
         # construct the subscription object and append it to zerver_subscription
         for member in channel['members']:
             sub = dict(
-                recipient=added_users[member],
+                recipient=stream_id_count, #proof : zerver/views/messages.py : L240, L324
```

Comment:
This reference to a code is mutable and the line range might change. You have to provide the link at a specific version.

Reply:
@rheaparekh this: https://github.com/zulip/zulip/blob/84bf6e3424d11e8fabc7980143e0639586f192ad/zerver/views/messages.py#L240-L324 ?

Reply:
Okay! it'll be https://github.com/zulip/zulip/blob/master/zerver/views/messages.py#L240 and https://github.com/zulip/zulip/blob/master/zerver/views/messages.py#L324. I couldn't find the exact documentation, but here we can see that type_id takes the value stream.id and user_profile.id. I'll add the links.